bpf/bpf_host.c

```diff
@@ -424,13 +435,20 @@ tail_handle_ipv6(struct __ctx_buff *ctx, const bool from_host)
 __section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_IPV6_FROM_HOST)
 int tail_handle_ipv6_from_host(struct __ctx_buff *ctx __maybe_unused)
 {
-	return tail_handle_ipv6(ctx, true);
+	__u32 ipcache_srcid = 0;
+
+#if defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6)
+	ipcache_srcid = ctx_load_meta(ctx, CB_IPCACHE_SRC_LABEL);
+	ctx_store_meta(ctx, CB_IPCACHE_SRC_LABEL, 0);
+#endif /* defined(ENABLE_HOST_FIREWALL) && !defined(ENABLE_MASQUERADE_IPV6) */
+
+	return tail_handle_ipv6(ctx, ipcache_srcid, true);
```

Comment:
This is still missing the corresponding write to `CB_IPCACHE_SRC_LABEL` in `do_netdev()`, correct?

Reply:
agree

Reply:
Hmm now the `CB_IPCACHE_SRC_LABEL` part is completely gone ... that's unexpected? Pretty sure we need it.